### PR TITLE
Ntrnl 524 update endpoints for the GitHub add collaborator page to match the new structure

### PR DESCRIPTION
**Pull request by DanielMurray97**
### JIRA link

Resolves [NTRNL-524](https://technologyprogramme.atlassian.net/browse/NTRNL-524?atlOrigin=eyJpIjoiNjYyOGQ0MDgzNDI4NGRhMGE4MjgwZWFhMDQzMDc2NmMiLCJwIjoiaiJ9)

### Description

- Update  the routing configuration to reflect the new endpoint structure for the collaborator page.
- Update views, controller, validation, and tests to reflect the end point changes.
- Update the validation mw function so it works with the new endpoint structure.

Screenshot from 2024-07-23 17-16-40

### Work checklist

- [x] Tests added where applicable
- [x] No vulnerability added

[NTRNL-524]: https://technologyprogramme.atlassian.net/browse/NTRNL-524?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/routes/collaborator.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Router } from 'express';
```
Comment (Mouhajer-CO) [inline src/routes/collaborator.ts:1]: Let's create a folder for `github` routing as we are going to have a few other paths, including `aws` and `dns`

File: src/validation/collaborator.validation.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { ErrorMessages } from './error.messages';
 import { githubHandleValidation } from './fields/github-handle.validation';
 import { repoNameValidation } from './fields/repo-name.validation';
 
-export const addCollaborator = [
```
Comment (Mouhajer-CO) [inline src/validation/collaborator.validation.ts:7]: Same here, let's create a folder for `github` validation

File: src/model/collaborator.model.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-export const AddCollaboratorKey = 'add_collaborator';
```
Comment (Mouhajer-CO) [inline src/model/collaborator.model.ts:1]: Same here, let's create a folder for github model

File: src/middleware/validation.middleware.ts [hunk 4]
```diff
@@ -13,13 +13,17 @@ export const checkValidations = (req: Request, res: Response, next: NextFunction
...
             const id = req.params[config.ID];
+
             // Removing trailing slash and 36 characters from UUID length
-            const template_path = (id) ? validatedFilepath.substring(0, validatedFilepath.length - 37).substring(1) : validatedFilepath.substring(1);
+            const templatePath = (id) ? validatedFilepath.substring(0, validatedFilepath.length - 37).substring(1) : validatedFilepath.substring(1);
             const errors = formatValidationError(errorList.array() as FieldValidationError[]);
 
-            log.info(`Validation error on ${template_path} page`);
+            // extracts the template name from the filepath
+            const template = templatePath.split('/').pop() as string;
```
Comment (Mouhajer-CO) [inline src/middleware/validation.middleware.ts:22]: Nice one 👍

File: test/integration/routes/collaborator.spec.ts [hunk 5]
```diff
@@ -14,35 +14,35 @@ import { authentication } from '../../../src/middleware/authentication.middlewar
 
 import {
```
Comment (Mouhajer-CO) [inline test/integration/routes/collaborator.spec.ts:15]: Same for the test directory, I would have created a folder for `github` routes and unit/controller

**Review by Mouhajer-CO — COMMENTED** [review]
Thanks @DanielMurray97 for the refactoring and validation update 🔥

File: src/middleware/validation.middleware.ts (hunk 4 above)
Comment (harley-harris) [inline src/middleware/validation.middleware.ts:22]: This is great @DanielMurray97 , will now work further nested templates. I'd suggest adding a test case in the `validation.middleware.spec` demonstrate this behaviour 👍

File: src/views/include/check-your-requests/collaborator.html [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 
-{% for collaborator in add_collaborator %}
+{% for collab in collaborator %}
 
-  {% set firstNameHref = "/add-collaborator/" + collaborator.id + previousPageQueryParam + "#first_name" %}
-  {% set lastNameHref = "/add-collaborator/" + collaborator.id + previousPageQueryParam + "#last_name" %}
-  {% set githubHandleHref = "/add-collaborator/" + collaborator.id + previousPageQueryParam + "#github_handle" %}
-  {% set emailAddressHref = "/add-collaborator/" + collaborator.id + previousPageQueryParam + "#email_address" %}
-  {% set repoHref = "/add-collaborator/" + collaborator.id + previousPageQueryParam + "#repo_name" %}
+  {% set firstNameHref = "/github/update/collaborator/" + collaborator.id + previousPageQueryParam + "#first_name" %}
```
Comment (Mouhajer-CO) [inline src/views/include/check-your-requests/collaborator.html:5]: `id` is undefined, variable name has to be change to `collab`. It should be `collab.id` some for the other reference

**Pull request merged**